.github/workflows/pull-request.yaml

```diff
@@ -0,0 +1,87 @@
+name: PR Check
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
+
+# Cancel any jobs in-progress for the current branch
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  id-token: write
+  contents: read
+  packages: read
+  repository-projects: read
```

Comment:
Not sure you need all these permissions for a PR workflow. If you do can you explain why?